### PR TITLE
[Rust] Added doc comments for Config

### PR DESCRIPTION
**Pull request by apepkuss**
This PR is to add the necessary descriptions for the Config struct.

### CODE REVIEW

File: bindings/rust/wasmedge-sys/src/config.rs [hunk 1]
```diff
@@ -1,79 +1,129 @@
...
+///
+///       Also see [Tail Call Proposal](https://github.com/WebAssembly/tail-call/blob/master/proposals/tail-call/Overview.md).
+///
+///       Also see [](https://github.com/WebAssembly/tail-call/blob/master/proposals/tail-call/Overview.md).
+///
+///     - `Annotations` supports annotations in WASM text format.
+///
+///       Also see [Annotations Proposal](https://github.com/WebAssembly/annotations/blob/master/proposals/annotations/Overview.md).
+///
+///     - `Memory64` support linear memory of sizes larger than 232 bits.
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/config.rs:58]: Why `232`?

File: bindings/rust/wasmedge-sys/src/config.rs [hunk 2]
```diff
@@ -95,99 +145,273 @@ impl Config {
...
     pub fn get_compiler_output_format(&self) -> CompilerOutputFormat {
         let value = unsafe { wasmedge::WasmEdge_ConfigureCompilerGetOutputFormat(self.ctx) };
         value.into()
     }
 
-    /// Set the dump IR boolean value of AOT compiler.
-    pub fn dump_ir(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureCompilerSetDumpIR(self.ctx, enable) };
+    /// Sets the dump IR option of AOT compiler.
+    pub fn enable_dump_ir(self, flag: bool) -> Self {
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/config.rs:219]: The `enable_` prefix is very confusing. Looks like this is just set the `dump_ir` option to `flag`.
I prefer to use `set` instead. So do the following `enable_`.

File: bindings/rust/wasmedge-sys/src/config.rs [hunk 3]
```diff
@@ -95,99 +145,273 @@ impl Config {
...
         unsafe { wasmedge::WasmEdge_ConfigureStatisticsIsTimeMeasuring(self.ctx) }
     }
 }
 
-// # TODO: WasmEdge_HostRegistration
+/// Code Generate for `WasmEdge_Proposal` :
+///
+/// ```rust
+/// impl Config {
+///     pub fn enable_bulkmemoryoperations(self, enable: bool) -> Self {
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/config.rs:278]: I am not sure if we should use `enable_` and `disable_` or `set_`. This is very confusing.

**Review by hydai — CHANGES_REQUESTED** [review]

File: bindings/rust/wasmedge-sys/src/config.rs (hunk 1 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/config.rs:58]: @hydai Accepted. It's a typo and should be power(2, 32). Thanks!

Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/config.rs:58]: Will refactor the statement. Thanks!

File: bindings/rust/wasmedge-sys/src/config.rs (hunk 2 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/config.rs:219]: Accepted. Will refactor the function name. Thanks!

File: bindings/rust/wasmedge-sys/src/config.rs (hunk 3 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/config.rs:278]: This comment and the subsequent macro were designed by other contributors. My suggestion is to keep them as they are now. After this round of the doc comment task is done, we plan to initiate a new task to review all APIs of Rust bindings, that could refine/remove some previously introduced design. Thanks!

File: bindings/rust/wasmedge-sys/src/config.rs (hunk 2 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/config.rs:219]: How about just using the original name `dump_ir`? It seems much better than `enable_dump_ir` and `set_dump_ir`.

Comment (hydai) [reply bindings/rust/wasmedge-sys/src/config.rs:219]: > How about just using the original name `dump_ir`? It seems much better than `enable_dump_ir` and `set_dump_ir`.

Agree. Simple and clear.

File: bindings/rust/wasmedge-sys/src/config.rs (hunk 3 above)
Comment (hydai) [reply bindings/rust/wasmedge-sys/src/config.rs:278]: Okay.

File: bindings/rust/wasmedge-sys/src/config.rs [hunk 4]
```diff
@@ -95,99 +145,273 @@ impl Config {
...
+    /// Checks if the dump IR option turns on or not.
     pub fn is_dump_ir(&self) -> bool {
         unsafe { wasmedge::WasmEdge_ConfigureCompilerIsDumpIR(self.ctx) }
     }
 
-    /// Set the generic binary option of AOT compiler.
-    pub fn generic_binary(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureCompilerSetGenericBinary(self.ctx, enable) };
+    /// Sets the generic binary option of AOT compiler.
+    pub fn enable_generic_binary(self, flag: bool) -> Self {
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/config.rs:230]: The following `enable_` prefix should also be removed.

File: bindings/rust/wasmedge-sys/src/config.rs [hunk 5]
```diff
@@ -95,99 +145,273 @@ impl Config {
...
+    /// Checks if the generic binary option of AOT compiler turns on or not.
     pub fn is_generic_binary(&self) -> bool {
         unsafe { wasmedge::WasmEdge_ConfigureCompilerIsGenericBinary(self.ctx) }
     }
 
-    /// Enable or disable instruction counting.
-    pub fn count_instructions(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureStatisticsSetInstructionCounting(self.ctx, enable) };
+    /// Sets the instruction counting option.
+    pub fn enable_instruction_counting(self, flag: bool) -> Self {
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/config.rs:241]: Ditto.

File: bindings/rust/wasmedge-sys/src/config.rs [hunk 6]
```diff
@@ -95,99 +145,273 @@ impl Config {
...
+    /// Checks if the instruction counting option turns on or not.
     pub fn is_instruction_counting(&self) -> bool {
         unsafe { wasmedge::WasmEdge_ConfigureStatisticsIsInstructionCounting(self.ctx) }
     }
 
-    /// Enable or disable cost cost measuring.
-    pub fn measure_costs(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureStatisticsSetCostMeasuring(self.ctx, enable) };
+    /// Sets the cost measuring option.
+    pub fn enable_cost_measuring(self, flag: bool) -> Self {
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/config.rs:252]: Ditto.

File: bindings/rust/wasmedge-sys/src/config.rs [hunk 7]
```diff
@@ -95,99 +145,273 @@ impl Config {
...
+    /// Checks if the cost measuring option turns on or not.
     pub fn is_cost_measuring(&self) -> bool {
         unsafe { wasmedge::WasmEdge_ConfigureStatisticsIsCostMeasuring(self.ctx) }
     }
 
-    /// Set the time measuring option.
-    pub fn measure_time(self, enable: bool) -> Self {
-        unsafe { wasmedge::WasmEdge_ConfigureStatisticsSetTimeMeasuring(self.ctx, enable) };
+    /// Sets the time measuring option.
+    pub fn enable_time_measuring(self, flag: bool) -> Self {
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/config.rs:263]: Ditto.

**Review by hydai — CHANGES_REQUESTED** [review]

File: bindings/rust/wasmedge-sys/src/config.rs (hunk 4 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/config.rs:230]: Accepted, thanks!

File: bindings/rust/wasmedge-sys/src/config.rs (hunk 5 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/config.rs:241]: Accepted, thanks!

File: bindings/rust/wasmedge-sys/src/config.rs (hunk 6 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/config.rs:252]: Accepted, thanks!

File: bindings/rust/wasmedge-sys/src/config.rs (hunk 7 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/config.rs:263]: Accepted, thanks!

**Comment by apepkuss** [comment]
Hi @hydai, the review issues have already been fixed, please review the latest changes again. In addition, the CI checks report a new failure, and the error message is shown below. It should not be caused by this fix. Thanks!
```
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/WasmEdge/www.git/'
```

**Review by hydai — APPROVED** [review]

**Pull request merged**